brainglobe_atlasapi/atlas_generation/atlas_scripts/perens_stereotaxic_mri_mouse.py

```diff
@@ -0,0 +1,402 @@
+        )
+    elif isinstance(key, int):
+        list_parent_ids = tree_traverse_child2parent(df, key, [])
+
+    if isinstance(key, str):  # if input is acronym convert IDs to acronyms
+        parents = []
+        for id in list_parent_ids:
+            parents.append(get_acronym_from_id(df, id))
+    elif isinstance(key, int):
+        parents = list_parent_ids.copy()
+
+    return parents
+
+
+##############################################################################
+
+##############################################################################
+# %%
+
+
+def create_atlas(working_dir, resolution):
+    ATLAS_NAME = "perens_stereotaxic_mri_mouse"
+    SPECIES = "Mus musculus"
+    ATLAS_LINK = "https://www.neuropedia.dk/resource/multimodal-3d-mouse-brain-atlas-framework-with-the-skull-derived-coordinate-system/"
+    CITATION = "Perens et al. 2023, https://doi.org/10.1007/s12021-023-09623-9"
+    ORIENTATION = "ial"
+    ROOT_ID = 997
+    ATLAS_FILE_URL = "https://www.neuropedia.dk/wp-content/uploads/Multimodal_mouse_brain_atlas_files.7z"
+
+    # Temporary folder for  download:
+    download_dir_path = working_dir / "downloads"
+    download_dir_path.mkdir(exist_ok=True)
+    atlas_files_dir = download_dir_path / "atlas_files"
+
+    ## Download atlas_file
+    utils.check_internet_connection()
+
+    import urllib.request
+
+    destination_path = download_dir_path / "atlas_download.7z"
+    headers = {
+        "User-Agent": (
+            "Mozilla/5.0 "
+            "(Windows NT 10.0; "
+            "Win64; x64; "
+            "rv:129.0) "
+            "Gecko/20100101 "
+            "Firefox/129.0"
+        ),
+        "Accept": (
+            "text/html,"
+            "application/xhtml+xml,"
+            "application/xml;q=0.9,"
+            "image/avif,"
+            "image/webp,"
+            "image/png,"
+            "image/svg+xml,"
+            "*/*;q=0.8"
+        ),
+        "Accept-Language": "en-GB,en;q=0.5",
+        "Accept-Encoding": "gzip, deflate, br, zstd",
+        "DNT": "1",
+        "Sec-GPC": "1",
+        "Host": "www.neuropedia.dk",
+        "Connection": "keep-alive",
+        "Upgrade-Insecure-Requests": "1",
+        "Sec-Fetch-Dest": "document",
+        "Sec-Fetch-Mode": "navigate",
+        "Sec-Fetch-Site": "none",
+        "Sec-Fetch-User": "?1",
+        "TE": "trailers",
+        "Priority": "u=0, i",
+    }
+    if not os.path.isdir(
+        atlas_files_dir / "Multimodal_mouse_brain_atlas_files"
+    ):
+        req = urllib.request.Request(ATLAS_FILE_URL, headers=headers)
+        with (
+            urllib.request.urlopen(req) as response,
+            open(destination_path, "wb") as out_file,
+        ):
+            data = response.read()  # a `bytes` object
+            out_file.write(data)
+        with py7zr.SevenZipFile(destination_path, mode="r") as z:
+            z.extractall(path=atlas_files_dir)
+        destination_path.unlink()
+
```

Comment:
`pooch` _should_ be able to take care of all of the below, and would make this considerably shorter, and more consistent with newer atlas packaging scripts (we aim to move all atlas packaging to `pooch` one day #284 ). See [`axolotl_atlas.py` for an example](https://github.com/brainglobe/brainglobe-atlasapi/blob/086ef0db7761ad3051d544d0228aa448f03246eb/brainglobe_atlasapi/atlas_generation/atlas_scripts/axolotl_atlas.py#L57C2-L64C10).

Reply:
Unfortunately it seems pooch cannot handle this as it requires specific headers (The gubra server has cloudflare protection which refuses connections that are not from a browser. so my workaround is to set the request headers to that of a browser). I tried to implement a similar solution in pooch with no luck.

Reply:
Great detective work, thanks - apologies for leading you down that rabbit hole!
I'd suggest encapsulating this part into a function, and document why we need this instead in its docstring?

Reply:
Great idea, done